azure-pipelines.yml

```diff
@@ -1341,6 +1278,169 @@ jobs:
+    displayName: Generate lks.json for .zip files
+    inputs:
+      PathtoPublish: $(Build.ArtifactStagingDirectory)/zipv2
+      ArtifactName: zipv2
+
+  - task: SdtReport@2
+    inputs:
+      GdnExportAllTools: true
+      GdnExportHtmlFile: true
+  - task: PublishSecurityAnalysisLogs@3
+    inputs:
+      ArtifactName: 'CodeAnalysisLogs'
+      ArtifactType: 'Container'
+      AllTools: true
+      ToolLogsNotFoundAction: 'Standard'
+  - task: PostAnalysis@2
+    inputs:
+      GdnBreakAllTools: true
+      GdnBreakFast: true
+
+- job: Phase_3
+  displayName: Run NON self-contained build for macos arm64
+  timeoutInMinutes: 120
+  cancelTimeoutInMinutes: 1
+  pool:
+    vmImage: macos-13-arm64
+  steps:
+  - checkout: self
+    clean: true
+    fetchTags: false
+    persistCredentials: True
+  - task: PowerShell@2
+    displayName: Set staging/prod build vars for signing and telemetry
+    inputs:
+      targetType: inline
+      script: >-
+        $branchName = $env:BUILD_SOURCEBRANCH
+
+        if ($branchName -eq "refs/heads/main") {
+            Write-Host "##vso[task.setvariable variable=TelemetryType;]${{ parameters.TelemetryType }}"
+            Write-Host "Done setting telemetry key instance as ${{ parameters.TelemetryType }} for main branch"
+        } else {
+            Write-Host "##vso[task.setvariable variable=TelemetryType;]${{ parameters.TelemetryType }}"
+            Write-Host "Done setting telemetry key instance as ${{ parameters.TelemetryType }} for other branch"
+        }
+
+        Write-Host "##vso[task.setvariable variable=MicroBuild_NuPkgSigningEnabled;]true"
+  - task: UseDotNet@2
+    displayName: Use .NET Core SDK 7.0.X
+    inputs:
+      version: $(BuildParameters.dotnetversion)
+      installationPath: $(Agent.TempDirectory)/dotnet
+  - task: DotNetCoreCLI@2
+    displayName: publish DSC for arm64 (OSX)
+    inputs:
+      command: publish
+      publishWebProjects: false
+      projects: src/dsc/dsc.csproj
+      arguments: -c ${{ parameters.BuildConfiguration }} -r osx-arm64 --self-contained false --verbosity detailed
+      zipAfterPublish: false
+      modifyOutputPath: false
+  - task: Bash@3
+    displayName: Download kubectl for OSX arm64
+    inputs:
+      targetType: 'inline'
+      script: |
+        mkdir -p "$(Build.SourcesDirectory)/src/dsc/bin/${{ parameters.BuildConfiguration }}/$(BuildParameters.dotnetversionforbuild)/osx-arm64/publish/kubectl/osx"
+        curl -L https://dl.k8s.io/release/v1.27.3/bin/darwin/arm64/kubectl -o "$(Build.SourcesDirectory)/src/dsc/bin/${{ parameters.BuildConfiguration }}/$(BuildParameters.dotnetversionforbuild)/osx-arm64/publish/kubectl/osx/kubectl"
+
+  - task: Bash@3
+    displayName: code sign for OSX arm64
+    inputs:
+      targetType: 'inline'
+      script: |
+        codesign --force --deep --sign - "$(Build.SourcesDirectory)/src/dsc/bin/${{ parameters.BuildConfiguration }}/$(BuildParameters.dotnetversionforbuild)/osx-arm64/publish/dsc"
```

Comment:
This is codesign part which is required for any arm64 mac's (M1) to execute any binaries.

Reply:
That sounds great! Does that mean that this PR supersedes https://github.com/Azure/vscode-bridge-to-kubernetes/pull/82 and makes that PR redundant?